### ISSUE TITLE
Client: Support for multiple result nodes

### ISSUE THREAD

**Issue by soxofaan** [Open-EO/openeo-python-client#391]
follow up of Open-EO/openeo-processes#401

python client needs some kind of support for building process graphs with multiple result nodes

**Comment by jdries** [comment]
It looks like we can create a valid process graph from multiple datacubes, as long as we use the same flattener:

```
week = s2cube.aggregate_temporal_period(period="week",reducer="median")
month = s2cube.aggregate_temporal_period(period="month",reducer="median")

cube1=week.save_result("netCDF")
cube2 = month.save_result("netCDF")

from openeo.internal.graph_building import GraphFlattener
flattener = GraphFlattener()
dict1 = flattener.flatten(cube1.result_node())
secondary_result = cube2.result_node()
dict2 = flattener.flatten(secondary_result)

final_graph = {"process_graph":dict1|dict2}
import json
merged = json.dumps(final_graph, indent=2)
print(merged)
```

**Comment by jdries** [comment]
so way forward is either updating create_job or a variant of that method that accepts multiple datacubes?

or
`cube.execute_batch(args..., extra_cubes=[extra_cube1])`
or
`connection.execute_batch([cube1,cube2]), other args...)`

**Comment by soxofaan** [comment]
In the python client we already have a base class `_ProcessGraphAbstraction`, which is further implemented by `DataCube`, `VectorCube`, `MlModel`. So I was thinking along the lines of creating another subclass of that, say `ProcessGraph` or `MultiResult`, which you can pass one or more cubes and work with that. E.g.

```python
results = MultiResult([cube_as_tiff, cube_as_netcdf, vector_cube])

results.execute_batch()
```

(all naming up for discussion of course)

**Comment by JeroenVerstraelen** [comment]
For grassland watch this is a requirement.

**Comment by soxofaan** [comment]
ready to merge:
- https://github.com/Open-EO/openeo-python-client/pull/651

just waiting for quick review if possible

**Comment by soxofaan** [comment]
merged #651
